### PR TITLE
PR Template: Added a little more guidance on how to structure PR desc…

### PR DESCRIPTION
**Pull request by fabiand**
### What this PR does
Before this PR there was less guidance on how to describe the scope of a PR

After this PR some more hints are left in the PR template to provide a better structure to described the changes of a PR

### Why we need it and why it was done in this way
The following tradeoffs were made: The goal was to document better WHY changes were made, and the reasoning history of them. Thus I added a few more quetsions, but still tried to keep it lean.

The following alternatives were considered: Keep it as is.

### Special notes for your reviewer

Please remember that this change will help people in future to understand the reasoning of PRs better

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [x] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [x] PR: The PR description is expressive enough and will help future contributors
- [x] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [x] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [x] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [x] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [x] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [x] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

## Release note

```release-note
NONE
```

### CODE REVIEW

**Comment by fabiand** [comment]
cc @AlonaKaplan @dhiller @vladikr @jean-edouard @xpivarc

**Comment by fabiand** [comment]
@vladikr @acardace @xpivarc any feedback?

**Review by dhiller — APPROVED** [review]
Makes sense!

**Comment by fabiand** [comment]
@aburdenthehand any feedback?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -3,14 +3,24 @@
 2. Follow the instructions for writing a release note from k8s: https://git.k8s.io/community/contributors/guide/release-notes.md
 -->
 
-**What this PR does / why we need it**:
+## What this PR does
+Before this PR
```
Comment (aburdenthehand) [inline .github/PULL_REQUEST_TEMPLATE.md:7]:
```suggestion
Before this PR: 
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -3,14 +3,24 @@
 2. Follow the instructions for writing a release note from k8s: https://git.k8s.io/community/contributors/guide/release-notes.md
 -->
 
-**What this PR does / why we need it**:
+## What this PR does
+Before this PR 
 
-**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
+After this PR
```
Comment (aburdenthehand) [inline .github/PULL_REQUEST_TEMPLATE.md:9]:
```suggestion
After this PR: 
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -3,14 +3,24 @@
...
-**Special notes for your reviewer**:
+## Why we need it and why it was done in this way
+The following tradeoffs were made:
+
+The following alternatives were considered:
+
+## Special notes for your reviewer
 
-**Checklist**
+None/Some
```
Comment (aburdenthehand) [inline .github/PULL_REQUEST_TEMPLATE.md:21]: Is the author meant to remove one or both of those? It's not clear to me.

**Review by aburdenthehand — COMMENTED** [review]
Makes sense (although not sure how big the headings will render). 
A couple of tiny suggestions and a question from me.
Thanks @fabiand !

**Comment by aburdenthehand** [comment]
I see now that you've used the heading markup in the description of this very PR. How delightful. 
I worry that level 2 heading is not too big. Would level 3 be better to consume less space and lessen the difference in size between heading and text?

**Comment by fabiand** [comment]
Adressed all of your comments, thanks for the review @aburden

**Review by dhiller — APPROVED** [review]
/approve

**Pull request merged**